src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/ValueContentAnalysis/ValueContentAbstractValue.cs

```diff
@@ -261,7 +261,7 @@ internal ValueContentAbstractValue IntersectLiteralValues(ValueContentAbstractVa
             }
 
             // Merge Literals
-            var builder = PooledHashSet<object?>.GetInstance();
+            using var builder = PooledHashSet<object?>.GetInstance();
```

Comment:
This will Dispose() on the error path (`return MaybeContainsNonLiteralState`) too.